core/arch/arm/plat-k3/drivers/ti_sci.c

```diff
@@ -213,6 +213,105 @@ int ti_sci_device_put(uint32_t id)
 	return ti_sci_device_set_state(id, 0, MSG_DEVICE_SW_STATE_AUTO_OFF);
 }
 
+int ti_sci_set_fwl_region(uint16_t fwl_id, uint16_t region,
+			  uint32_t n_permission_regs, uint32_t control,
+			  uint32_t permissions[FWL_MAX_PRIVID_SLOTS],
+			  uint64_t start_address, uint64_t end_address)
+{
+	struct ti_sci_msg_req_fwl_set_firewall_region req = { };
+	struct ti_sci_msg_resp_fwl_set_firewall_region resp = { };
+	struct ti_sci_xfer xfer = { };
+	int ret = 0;
+
+	ret = ti_sci_setup_xfer(TI_SCI_MSG_FWL_SET, 0,
+				&req, sizeof(req),
+				&resp, sizeof(resp),
+				&xfer);
+	if (ret)
+		return ret;
+
+	req.fwl_id = fwl_id;
+	req.region = region;
+	req.n_permission_regs = n_permission_regs;
+	req.control = control;
+	req.permissions[0] = permissions[0];
```

Comment:
Wouldn't it make sense to use `memcmp(req.permissions, permissions, sizeof(req.permissions))` instead?

Reply:
Yeah, it's not a big deal. My main concern is that `FWL_MAX_PRIVID_SLOTS` is used above, but here we rely on the knowledge that `FWL_MAX_PRIVID_SLOTS` always will be `3`.

If wrote this code I would have left a `static_assert(FWL_MAX_PRIVID_SLOTS == 3)` around here. Do as you like.

Reply:
`FWL_MAX_PRIVID_SLOTS` will always be 3, I just make it a define to avoid using 3 as a magic number where I can, I'll add that `static_assert` for sanity sake.

Reply:
Hmm, looking at it, since `n_permission_regs` defines the amount of these slots we set, I can just loop on that. Plus a check for `n_permission_regs <= FWL_MAX_PRIVID_SLOTS` and I think we solve both issues.